golos.social/golos.social.cpp

```diff
@@ -15,16 +15,9 @@ void social::pin(name pinner, name pinning) {
 
     tables::pinblock_table table(_self, pinner.value);
     auto itr = table.find(pinning.value);
-    bool item_exists = (itr != table.end());
-
-    if (item_exists) {
+    if (itr != table.end()) {
         eosio::check(!itr->blocking, "You have blocked this account. Unblock it before pinning");
         eosio::check(!itr->pinning, "You already have pinned this account");
-
```

Comment:
What will happen if two checks are successful? I understand that this is not possible with correct entries. But if there is a record in the database, then the action will be performed, but no effect will occur. It is worth in this case at least to assert (such a situation is impossible, an error in the data).
And it turns out that the `pin`/`block` behavior is different in this case.

Reply:
@s-medvedev Added `SYSTEM` `eosio::check`s.